src/VisualStudio/Core/Def/SampleToolWindow/DocumentOutlineHelper.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.VisualStudio.LanguageServices
+{
+    internal static class DocumentOutlineHelper
+    {
+        internal static List<DocSymbol> GetDocumentSymbols(DocumentSymbol[]? body)
+        {
+            var docSymbols = new List<DocSymbol>();
+            if (body is not null && body.Length > 0)
```

Comment:
💡 we could check for null and zero length short-circuit to returning an empty list instead of doing the work

Reply:
Done